### PR TITLE
BM player addition

### PR DESCRIPTION
**Pull request by GGOUSSEAUD**

### CODE REVIEW

**Comment by GGOUSSEAUD** [comment]
Fixed everything, everything should be ready to merge!

File: axelrod/strategies/bm.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
+import random
+from axelrod.actions import Actions, Action
+from axelrod.player import Player
+
+C, D = Actions.C, Actions.D
+
+
+class BM(Player):
+    """
+    A player that is based on Bush Mosteller reinforced learning algorithm, he decides what he'll
```
Comment (meatballs) [inline axelrod/strategies/bm.py]: Could we please use gender neutral pronouns? - 'it' rather than 'he'

File: axelrod/strategies/bm.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    play only depending on his own previous payoffs.
```
Comment (meatballs) [inline axelrod/strategies/bm.py]: 'its' not 'his'

File: axelrod/strategies/bm.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+    The probability of playing C or D will be updated using a stimulus which represents
+    a win or a loss of value based on his previous play's payoff in the specified probability.
```
Comment (meatballs) [inline axelrod/strategies/bm.py]: 'its' not 'his'

File: axelrod/strategies/bm.py [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    The more a play will be rewarded through rounds , the more the player will be tempted to use it.
```
Comment (meatballs) [inline axelrod/strategies/bm.py]: typo: remove the space before the comma

File: axelrod/strategies/bm.py [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+        """
+        super().__init__()
+        self._c_probability , self._d_probability = c_probability , d_probability
+        self._init_c_probability , self._init_d_probability = c_probability , d_probability 
+        self._aspiration_level = abs((max(self.match_attributes["game"].RPST()) / aspiration_level_divider))
+        self._stimulus = 0.0
+        self._learning_rate = learning_rate
+
+    
+    def stimulus_update(self, opponent: Player):
```
Comment (meatballs) [inline axelrod/strategies/bm.py]: This method needs a docstring

File: axelrod/strategies/bm.py [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+        if self.history[-1] == D:
+           
+            if self._stimulus >= 0:
+                self._d_probability = self._d_probability + self._learning_rate * self._stimulus * (1 - self._d_probability)
+           
+            elif self._stimulus < 0:
+                self._d_probability = self._d_probability + self._learning_rate * self._stimulus * self._d_probability
+    
+
+    def BM_random_choice(self) -> Action:
```
Comment (meatballs) [inline axelrod/strategies/bm.py]: docstring

File: axelrod/tests/strategies/test_bm.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+from .test_player import TestPlayer
+
+C, D = axelrod.Actions.C, axelrod.Actions.D
+
+class TestBM(TestPlayer):
+    """
+    Note that this test is referred to in the documentation as an example on
+    writing tests.  If you modify the tests here please also modify the
+    documentation.
+    """
```
Comment (meatballs) [inline axelrod/tests/strategies/test_bm.py]: Please remove this docstring. It doesn't apply here

**Comment by meatballs** [comment]
@GGOUSSEAUD There are a lot of PEP8 errors in `bm.py` which need to be fixed.

Shout on gitter if you need help with that.

**Comment by GGOUSSEAUD** [comment]
Thank you for your review, I'll patch everything that is needed when I have time, right back at you when it's done.

**Comment by GGOUSSEAUD** [comment]
Gotcha ! I'll fix this when I have some spare time. I'm having a tough time with PEP 8 , I get a lot of errors x)

**Comment by drvinceknight** [comment]
@GGOUSSEAUD and @meatballs in 72ba385 I've made the minor adjustments:

- Improving docstrings;
- PEP8 improvements;
- Removing unneeded reset
- Using the in built random choice
- Changing the name

There might well be other things (it's been a while since I looked at this PR) that are still needed. 

(I've also fixed merge conflicts and rebased on to master, as it's been a while the diff was horrendous other wise :) :+1:.)

**Comment by drvinceknight** [comment]
@marcharper: @meatballs and I have gone through and OK'd this but I did the last couple of things and made minor adjustments like memory length etc. If you could have a quick check and OK that'd be great :+1:

**Pull request merged**